slatedb/src/compactor_state.rs

```diff
@@ -56,14 +56,90 @@ impl SourceId {
 pub(crate) enum CompactionStatus {
     Submitted,
     #[allow(dead_code)]
+    Pending,
+    #[allow(dead_code)]
     InProgress,
+    #[allow(dead_code)]
+    Completed,
+    #[allow(dead_code)]
+    Failed,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub(crate) enum CompactionType {
+    Internal,
+    #[allow(dead_code)]
+    External,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+/// In-memory description of the inputs to a compaction.
```

Comment:
Nit: For `///` rust docs, mind putting them just above the `#[]`  macros? We've been using this convention in the codebase.